libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1087,10 +1089,20 @@ bool TypeChecker::visit(VariableDeclarationStatement const& _statement)
 	{
 		// No initial value is only permitted for single variables with specified type.
 		if (_statement.declarations().size() != 1 || !_statement.declarations().front())
-			m_errorReporter.fatalTypeError(_statement.location(), "Assignment necessary for type detection.");
+		{
+			// Bailing out *fatal* here, as those (untyped) vars may be used later, and diagnostics wouldn't be helpful then.
+			if (!boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
+				m_errorReporter.fatalTypeError(_statement.location(), "Use of the \"var\" keyword is disallowed.");
+			else
+				// it is okay to not just return here, as there are no named components on the
+				// left-hand-side that could cause any damage later.
```

Comment:
I think the idea in the earlier review comment was to use ``else { solAssert(m_errorReporter.hasErrors(), ""); return false; }`` instead of just ``else return false;`` here to make sure that the syntax checker indeed already reported an error before.

Reply:
I'm not sure anymore why it didn't work when I've done the solAssert, but now everything is working as expected. Thx.